src/IncludedCommands/Resources/PersistentBans.luau

```diff
@@ -149,20 +152,128 @@ end
 
 --[[
 Bans a user id.
+
+Preserved for backwards compatibility and minimal script changes.
+--]]
+function PersistentBans.BanId(self: PersistentBans, UserId: number, Message: string | boolean, ModeratorId: number?, ModeratorName: string?): (boolean, string, {number}, {number})
+    return self:BanIds({
+        UserIds = {UserId},
+        Duration = -1,
+        DisplayReason = type(Message) == "string" and Message or "No ban reason specified.",
+        PrivateReason = type(Message) == "string" and Message or "No ban reason specified.",
+        ExcludeAltAccounts = false,
+        ApplyToUniverse = true
+    }, ModeratorId or 0, ModeratorName or "NexusAdmin")
+end
+
+--[[
+Bans user ids.
 --]]
-function PersistentBans.BanId(self: PersistentBans, UserId: number, Message: string?): ()
-    if not self.BansDataStore then return end
-    self.BansDataStore:Set(tostring(UserId), Message or true)
+function PersistentBans.BanIds(self: PersistentBans, Config: BanConfigType, ModeratorId: number?, ModeratorName: string?): (boolean, string, {number}, {number})
+    local CanBan, Immune = self:FilterIds(Config.UserIds, ModeratorId)
```

Comment:
`ModeratorId` is `number?` but `FilterIds` expects `number`, and is required. I'm not sure if this needs to be `ModetorId or 0`, or something else.